src/accountant_skel.rs

```diff
@@ -198,16 +199,21 @@ impl<W: Write + Send + 'static> AccountantSkel<W> {
 
     fn process(
         obj: &Arc<Mutex<AccountantSkel<W>>>,
-        verified_receiver: &Receiver<(Vec<SharedPackets>, Vec<Vec<u8>>)>,
+        verified_receiver: &Receiver<Vec<(SharedPackets, Vec<u8>)>>,
         blob_sender: &streamer::BlobSender,
         packet_recycler: &packet::PacketRecycler,
         blob_recycler: &packet::BlobRecycler,
     ) -> Result<()> {
         let timer = Duration::new(1, 0);
-        let (mms, vvs) = verified_receiver.recv_timeout(timer)?;
-        for (msgs, vers) in mms.into_iter().zip(vvs.into_iter()) {
+        let mms = verified_receiver.recv_timeout(timer)?;
+        for (msgs, vers) in mms {
             let reqs = Self::deserialize_packets(&msgs.read().unwrap());
-            let rsps = Self::process_packets(obj, reqs, vers);
+            let req_vers = reqs.into_iter()
+                .zip(vers)
+                .filter_map(|(req, ver)| req.map(|(msg, addr)| (msg, addr, ver)))
+                .filter(|x| x.0.verify())
+                .collect();
```

Comment:
this is going to be one big alloc

Reply:
It's using into_iter(). Should reduce to memmove().

Reply:
Vec<Vec<>> into Vec, right?  So the output is going to be all of the requests we just pulled up.  basically this will kill the improvement we are trying to get with the allocator, since it will allocate all the space for all the IO buffers copy into it, and them drop that buffer later.

Reply:
or do you mean its smart enough to know about the into_iter on line 163 and compile the allocation out?